### PR TITLE
Introduce source containers

### PR DESCRIPTION
**Pull request by MartinBasti**
* OSBS-8046

Signed-off-by: Martin Bašti <mbasti@redhat.com>

### CODE REVIEW

File: docs/users.rst [hunk 1]
```diff
@@ -1170,3 +1170,70 @@ stored in koji build, in section ``build.extra.operator_manifests.appregistry``.
 
 Manifests will not be pushed to the application registry for scratch builds,
 isolated builds, or re-builds to prevent unwanted changes
+
+
+Building Source Container Images
+================================
+
+OSBS is able to build source container image from a particular koji build
+previously created by OSBS. To crate a source container build you have to
```
Comment (athos-ribeiro) [inline docs/users.rst]: s/crate/create/

File: docs/users.rst [hunk 2]
```diff
@@ -1170,3 +1170,70 @@ stored in koji build, in section ``build.extra.operator_manifests.appregistry``.
...
+Under the hood the BSI_ project is used to generate source images from
+sources identified and collected by OSBS.
+
+.. _`BSI`: https://github.com/containers/BuildSourceImage
+
+
+Koji integration
+----------------
+
+Koji integration must be enabled for building source container images.
```
Comment (athos-ribeiro) [inline docs/users.rst]: Could we mention the reason for needing koji here?

File: docs/users.rst [hunk 3]
```diff
@@ -1170,3 +1170,70 @@ stored in koji build, in section ``build.extra.operator_manifests.appregistry``.
...
 Manifests will not be pushed to the application registry for scratch builds,
 isolated builds, or re-builds to prevent unwanted changes
+
+
+Building Source Container Images
+================================
+
+OSBS is able to build source container image from a particular koji build
+previously created by OSBS. To crate a source container build you have to
+specify either koji N-V-R or koji build ID.
```
Comment (athos-ribeiro) [inline docs/users.rst]: How about
... koji build ID for the image build you want to create a source container image for.

**Comment by athos-ribeiro** [comment]
It would also be nice to add a section on how the signing intents are picked up in order to fetch packages. See https://github.com/containerbuildsystem/atomic-reactor/blob/master/atomic_reactor/plugins/pre_fetch_sources.py#L200 for reference

**Comment by MartinBasti** [comment]
Comments addressed, thank you

File: docs/users.rst [hunk 4]
```diff
@@ -1170,3 +1170,95 @@ stored in koji build, in section ``build.extra.operator_manifests.appregistry``.
...
+
+OSBS is able to build source container image from a particular koji build
+previously created by OSBS. To create a source container build you have to
+specify either koji N-V-R or build ID for the image build you want to
+create a source container image for.
+
+
+Under the hood the BSI_ project is used to generate source images from
+sources identified and collected by OSBS.
+
```
Comment (athos-ribeiro) [inline docs/users.rst]: Do we want to mention that the BSI script should be available as `bsi` in the buildroot $PATH?

File: docs/users.rst [hunk 5]
```diff
@@ -1170,3 +1170,95 @@ stored in koji build, in section ``build.extra.operator_manifests.appregistry``.
...
+Building source container images using koji
+-------------------------------------------
+
+Using a koji client CLI directly you have to specify git repo URL and branch::
+
+    koji source-container-build <target>  --koji-build-nvr=NVR --koji-build-id=ID
+
+For a full list of options::
+
+    koji container-build --help
```
[the remark this answers is not in the document]
Comment (MartinBasti) [reply docs/users.rst]: fixed

File: docs/users.rst (hunk 4 above)
Comment (MartinBasti) [reply docs/users.rst]: added

File: docs/users.rst [hunk 6]
```diff
@@ -1170,3 +1170,96 @@ stored in koji build, in section ``build.extra.operator_manifests.appregistry``.
...
+* only koji internal RPMs are supported
+Support for other types of sources and external builds will be added in future.
+
+.. _`BSI`: https://github.com/containers/BuildSourceImage
+
+
+Signing intent resolution
+-------------------------
+
+Resolution of singing intent is done in following order:
```
Comment (twaugh) [inline docs/users.rst]: `singing` → `signing`

Comment (MartinBasti) [reply docs/users.rst]: :microphone: fixed :notes:

**Pull request merged**